lib/src/searchfield.dart

```diff
@@ -752,6 +751,7 @@ class _SearchFieldState<T> extends State<SearchField<T>> {
       filteredResult.addAll(widget.suggestions);
       // hide the suggestions
       suggestionStream.sink.add(null);
+      searchController!.text = item.searchKey;
```

Comment:
We don't need to set selected Value in searchfield this has already been taken care of on line 724, So you can remove it